pkg/segment/results/segresults/segresults.go

```diff
@@ -268,9 +268,25 @@ func (sr *SearchResults) UpdateNonEvalSegStats(runningSegStat *structs.SegStats,
 	var err error
 	switch measureAgg.MeasureFunc {
 	case utils.Min:
-		sstResult, err = segread.GetSegMin(runningSegStat, incomingSegStat)
+		res, err := segread.GetSegMin(runningSegStat, incomingSegStat)
+		if err != nil {
+			return nil, fmt.Errorf("UpdateSegmentStats: error getting segment level stats for %v, err: %v, qid=%v", measureAgg.String(), err, sr.qid)
+		}
+		sr.segStatsResults.measureResults[measureAgg.String()] = *res
+		if runningSegStat == nil {
+			return incomingSegStat, nil
+		}
+		return runningSegStat, nil
```

Comment:
nit: instead of
```
		if runningSegStat == nil {
			return incomingSegStat, nil
		}
		return runningSegStat, nil
```
can you do `return res, nil`? Or will that not be the same return value?

Reply:
No, res and runningSegStat are different things. res is just the finalMeasureValue.